### ISSUE TITLE
Support for OSX Command Line Tools without complete Xcode installation

### ISSUE THREAD

**Issue by gipert** [JuliaInterop/Cxx.jl#363]
I tried to `import Cxx`:
```julia
julia> import Cxx
INFO: Precompiling module Cxx.
ERROR: LoadError: LoadError: Could not find C++ standard library. Is XCode installed?
```
And then realised that this package relies on the command line tools shipped with XCode. However, for OSX users that do not use Xcode (and thus do not want to install such a big software), it is possible to only install the Command Line Tools with
```shell
$ xcode-select --install
```
in `/Library/Developer/CommandLineTools`. In fact, the `xcodebuild` command is not available:
```shell
$ xcodebuild -version
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```
However:
```shell
$ c++ -v
Apple LLVM version 9.0.0 (clang-900.0.38)
Target: x86_64-apple-darwin17.2.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```
Where you aware of this?

**Comment by EmileParolin** [comment]
Hi,
I found a workaround that does the job on my machine, using just the Command Line Tools
Just replace line [27](https://github.com/Keno/Cxx.jl/blob/f6dd8baafbf6ccf6e8ffdab9f76b7a82c2eac8d9/src/initialization.jl#L27) in [src/initialization.jl](https://github.com/Keno/Cxx.jl/blob/f6dd8baafbf6ccf6e8ffdab9f76b7a82c2eac8d9/src/initialization.jl) by
`
sysroot = @static is_apple() ? strip(readstring('xcrun --sdk macosx --show-sdk-path')) : C_NULL
`
and line [218](https://github.com/Keno/Cxx.jl/blob/f6dd8baafbf6ccf6e8ffdab9f76b7a82c2eac8d9/src/initialization.jl#L218) by
`
 "/Library/Developer/CommandLineTools/"
`
Hope that helps.

**Comment by gipert** [comment]
Hey @EmileParolin, thanks for the `xcrun` tip! I made some tests and it works nicely either with the full Xcode and with only the Command Line Tools. Also, I found that you can use `xcode-select -p` instead of hard-coding the path at line 218, I'll prepare a PR for that!

**Comment by gipert** [comment]
Even if it's not that perfect, while w/o Xcode the `xcode-select -p` command returns `/Library/Developer/CommandLineTools/`, w/ Xcode it returns only `/Applications/Xcode.app/Contents/Developer`, which is only part of the complete path...

**Comment by Gnimuc** [comment]
I think this has been fixed by #364.
